Compilation on OpenBSD/adJ 6.2

# Basic information
  -  Type of CPU: amd64

# Compile issues
  - Operating system: OpenBSD/adJ 6.2

It is possible to start building with:
```sh
cmake . -DCUDA_ENABLE=OFF -DOpenCL_ENABLE=OFF -DMICROHTTPD_ENABLE=OFF -DHWLOC_ENABLE=OFF
```
But runing `make` produces:
```
[ 44%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/backend/cpu/minethd.cpp.o
/home/vtamara/comp/monero/xmr-stak/xmrstak/backend/cpu/minethd.cpp:88:2: error: unknown type name 'cpu_set_t'
        cpu_set_t mn;
        ^
/home/vtamara/comp/monero/xmr-stak/xmrstak/backend/cpu/minethd.cpp:91:42: error: use of undeclared identifier 'cpu_set_t'
        return pthread_setaffinity_np(h, sizeof(cpu_set_t), &mn) == 0;
                                                ^
2 errors generated.
```

Issue Compiling STAK
Which OS do you use? OpenBSD/adJ 6.2

What software are you using to compile? Include versions (this can be very important).

$ clang -v
OpenBSD clang version 4.0.0 (tags/RELEASE_400/final) (based on LLVM 4.0.0)
Target: amd64-unknown-openbsd6.2
Thread model: posix
InstalledDir: /usr/bin

Add all commands you used and the full compile output here


```
$cmake . -DCUDA_ENABLE=OFF -DOpenCL_ENABLE=OFF -DMICROHTTPD_ENABLE=OFF -DHWLOC_ENABLE=OFF
...
$ make
Scanning dependencies of target xmr-stak-c
[  4%] Building C object CMakeFiles/xmr-stak-c.dir/xmrstak/backend/cpu/crypto/c_blake256.c.o
[  8%] Building C object CMakeFiles/xmr-stak-c.dir/xmrstak/backend/cpu/crypto/c_groestl.c.o
[ 12%] Building C object CMakeFiles/xmr-stak-c.dir/xmrstak/backend/cpu/crypto/c_jh.c.o
[ 16%] Building C object CMakeFiles/xmr-stak-c.dir/xmrstak/backend/cpu/crypto/c_keccak.c.o
[ 20%] Building C object CMakeFiles/xmr-stak-c.dir/xmrstak/backend/cpu/crypto/c_skein.c.o
[ 24%] Linking C static library bin/libxmr-stak-c.a
[ 24%] Built target xmr-stak-c
Scanning dependencies of target xmr-stak-backend
[ 28%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/jconf.cpp.o
[ 32%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/version.cpp.o
[ 36%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/backend/cpu/hwlocMemory.cpp.o
[ 40%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/backend/cpu/jconf.cpp.o
[ 44%] Building CXX object CMakeFiles/xmr-stak-backend.dir/xmrstak/backend/cpu/minethd.cpp.o
/home/vtamara/comp/monero/xmr-stak-upstream/xmrstak/backend/cpu/minethd.cpp:88:2: error: unknown type name 'cpu_set_t'
        cpu_set_t mn;
        ^
/home/vtamara/comp/monero/xmr-stak-upstream/xmrstak/backend/cpu/minethd.cpp:91:42: error: use of undeclared identifier 'cpu_set_t'
        return pthread_setaffinity_np(h, sizeof(cpu_set_t), &mn) == 0;
                                                ^
2 errors generated.
*** Error 1 in . (CMakeFiles/xmr-stak-backend.dir/build.make:159 'CMakeFiles/xmr-stak-backend.dir/xmrstak/backend/cpu/minethd.cpp.o':
/usr/b...)
*** Error 1 in . (CMakeFiles/Makefile2:68 'CMakeFiles/xmr-stak-backend.dir/all')
*** Error 1 in /home/vtamara/comp/monero/xmr-stak-upstream (Makefile:128 'all')

```

Run cmake -LA . in the build folder and add the output here

```
$ cmake -LA .                               
-- Set miner currency to 'monero' and 'aeon'
-- Configuring done                                    
-- Generating done                                       
-- Build files have been written to: /home/vtamara/comp/monero/xmr-stak-upstream
-- Cache values                                          
CMAKE_AR:FILEPATH=/usr/bin/ar                                  
CMAKE_BUILD_TYPE:STRING=Release                        
CMAKE_COLOR_MAKEFILE:BOOL=ON                                 
CMAKE_CXX_COMPILER:FILEPATH=/usr/bin/c++
CMAKE_CXX_FLAGS:STRING= 
CMAKE_CXX_FLAGS_DEBUG:STRING=-g                  
CMAKE_CXX_FLAGS_MINSIZEREL:STRING=-DNDEBUG
CMAKE_CXX_FLAGS_RELEASE:STRING=-DNDEBUG     
CMAKE_CXX_FLAGS_RELWITHDEBINFO:STRING=-g -DNDEBUG
CMAKE_C_COMPILER:FILEPATH=/usr/bin/cc                     
CMAKE_C_FLAGS:STRING=                
CMAKE_C_FLAGS_DEBUG:STRING=-g                       
CMAKE_C_FLAGS_MINSIZEREL:STRING=-DNDEBUG
CMAKE_C_FLAGS_RELEASE:STRING=-DNDEBUG              
CMAKE_C_FLAGS_RELWITHDEBINFO:STRING=-g -DNDEBUG
CMAKE_EXE_LINKER_FLAGS:STRING=
CMAKE_EXE_LINKER_FLAGS_DEBUG:STRING=              
CMAKE_EXE_LINKER_FLAGS_MINSIZEREL:STRING=                
CMAKE_EXE_LINKER_FLAGS_RELEASE:STRING=               
CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO:STRING=                                                                         
CMAKE_EXPORT_COMPILE_COMMANDS:BOOL=OFF
CMAKE_INSTALL_PREFIX:PATH=/home/vtamara/comp/monero/xmr-stak-upstream
CMAKE_LINKER:FILEPATH=/usr/bin/ld                                                                                                 
CMAKE_LINK_STATIC:BOOL=OFF                                            
CMAKE_MAKE_PROGRAM:FILEPATH=/usr/bin/make        
CMAKE_MODULE_LINKER_FLAGS:STRING=
CMAKE_MODULE_LINKER_FLAGS_DEBUG:STRING=                                                                                              
CMAKE_MODULE_LINKER_FLAGS_MINSIZEREL:STRING=
CMAKE_MODULE_LINKER_FLAGS_RELEASE:STRING=                                       
CMAKE_MODULE_LINKER_FLAGS_RELWITHDEBINFO:STRING=                               
CMAKE_NM:FILEPATH=/usr/bin/nm                    
CMAKE_OBJCOPY:FILEPATH=/usr/bin/objcopy     
CMAKE_OBJDUMP:FILEPATH=/usr/bin/objdump                
CMAKE_RANLIB:FILEPATH=/usr/bin/ranlib                    
CMAKE_SHARED_LINKER_FLAGS:STRING=                                               
CMAKE_SHARED_LINKER_FLAGS_DEBUG:STRING=                  
CMAKE_SHARED_LINKER_FLAGS_MINSIZEREL:STRING=                   
CMAKE_SHARED_LINKER_FLAGS_RELEASE:STRING=              
CMAKE_SHARED_LINKER_FLAGS_RELWITHDEBINFO:STRING=             
CMAKE_SKIP_INSTALL_RPATH:BOOL=NO        
CMAKE_SKIP_RPATH:BOOL=NO
CMAKE_STATIC_LINKER_FLAGS:STRING=                
CMAKE_STATIC_LINKER_FLAGS_DEBUG:STRING=   
CMAKE_STATIC_LINKER_FLAGS_MINSIZEREL:STRING=
CMAKE_STATIC_LINKER_FLAGS_RELEASE:STRING=        
CMAKE_STATIC_LINKER_FLAGS_RELWITHDEBINFO:STRING=          
CMAKE_STRIP:FILEPATH=/usr/bin/strip  
CMAKE_VERBOSE_MAKEFILE:BOOL=FALSE                   
CPU_ENABLE:BOOL=ON                      
CUDA_64_BIT_DEVICE_CODE:BOOL=ON                    
CUDA_ATTACH_VS_BUILD_RULE_TO_CUDA_FILE:BOOL=ON 
CUDA_BUILD_CUBIN:BOOL=OFF     
CUDA_BUILD_EMULATION:BOOL=OFF                     
CUDA_CUDART_LIBRARY:FILEPATH=CUDA_CUDART_LIBRARY-NOTFOUND
CUDA_CUDA_LIBRARY:FILEPATH=CUDA_CUDA_LIBRARY-NOTFOUND
CUDA_ENABLE:BOOL=OFF                                         
CUDA_GENERATED_OUTPUT_DIR:PATH=       
CUDA_HOST_COMPILATION_CPP:BOOL=ON                                    
CUDA_HOST_COMPILER:FILEPATH=/usr/bin/cc                                                                                           
CUDA_NVCC_EXECUTABLE:FILEPATH=CUDA_NVCC_EXECUTABLE-NOTFOUND           
CUDA_NVCC_FLAGS:STRING=                          
CUDA_NVCC_FLAGS_DEBUG:STRING=    
CUDA_NVCC_FLAGS_MINSIZEREL:STRING=                                                                                                   
CUDA_NVCC_FLAGS_RELEASE:STRING=             
CUDA_NVCC_FLAGS_RELWITHDEBINFO:STRING=                                          
CUDA_PROPAGATE_HOST_FLAGS:BOOL=ON                                              
CUDA_SDK_ROOT_DIR:PATH=CUDA_SDK_ROOT_DIR-NOTFOUND
CUDA_SEPARABLE_COMPILATION:BOOL=OFF         
CUDA_TOOLKIT_INCLUDE:PATH=CUDA_TOOLKIT_INCLUDE-NOTFOUND
CUDA_TOOLKIT_ROOT_DIR:PATH=CUDA_TOOLKIT_ROOT_DIR-NOTFOUND
CUDA_VERBOSE_BUILD:BOOL=OFF                                                     
CUDA_cublas_LIBRARY:FILEPATH=CUDA_cublas_LIBRARY-NOTFOUND
CUDA_cublasemu_LIBRARY:FILEPATH=CUDA_cublasemu_LIBRARY-NOTFOUND
CUDA_cufft_LIBRARY:FILEPATH=CUDA_cufft_LIBRARY-NOTFOUND
CUDA_cufftemu_LIBRARY:FILEPATH=CUDA_cufftemu_LIBRARY-NOTFOUND
HWLOC:FILEPATH=HWLOC-NOTFOUND           
HWLOC_ENABLE:BOOL=OFF   
HWLOC_INCLUDE_DIR:PATH=HWLOC_INCLUDE_DIR-NOTFOUND
MHTD:FILEPATH=MHTD-NOTFOUND               
MICROHTTPD_ENABLE:BOOL=OFF                  
MTHD_INCLUDE_DIR:PATH=MTHD_INCLUDE_DIR-NOTFOUND  
OPENSSL_CRYPTO_LIBRARY:FILEPATH=/usr/lib/libcrypto.so.42.0
OPENSSL_INCLUDE_DIR:PATH=/usr/include
OPENSSL_SSL_LIBRARY:FILEPATH=/usr/lib/libssl.so.44.1
OpenCL_ENABLE:BOOL=OFF                  
OpenCL_INCLUDE_DIR:PATH=OpenCL_INCLUDE_DIR-NOTFOUND
OpenCL_LIBRARY:FILEPATH=OpenCL_LIBRARY-NOTFOUND
OpenSSL_ENABLE:BOOL=ON        
PKG_CONFIG_EXECUTABLE:FILEPATH=/usr/bin/pkg-config
XMR-STAK_COMPILE:STRING=native                           
XMR-STAK_CURRENCY:STRING=all                         
```
Describe the issue clearly:

a. A short synopsis of what is happening.

Cannot complie because there is no `function pthread_setaffinity_np` in OpenBSD.

b. What you expect should be happening.

Compile

c. Exact error message(s), if any.

/home/vtamara/comp/monero/xmr-stak-upstream/xmrstak/backend/cpu/minethd.cpp:91:42: error: use of undeclared identifier 'cpu_set_t'
        return pthread_setaffinity_np(h, sizeof(cpu_set_t), &mn) == 0;


d. Include screenshots and/or logs of the issue.

See above

e. If useful, edit the screenshot with circles or arrows to highlight the error.

Describe what steps you have performed in an attempt to resolve the issue.

I could compile with the change proposed in the pull request https://github.com/fireice-uk/xmr-stak/pull/772

--- a/xmrstak/backend/cpu/crypto/cryptonight_common.cpp
+++ b/xmrstak/backend/cpu/crypto/cryptonight_common.cpp
@@ -253,6 +253,9 @@ cryptonight_ctx* cryptonight_alloc_ctx(size_t use_fast_mem, size_t use_mlock, al
 #elif defined(__FreeBSD__)
 	ptr->long_state = (uint8_t*)mmap(0, hashMemSize, PROT_READ | PROT_WRITE,
 		MAP_PRIVATE | MAP_ANONYMOUS | MAP_ALIGNED_SUPER | MAP_PREFAULT_READ, -1, 0);
+#elif defined(__OpenBSD__)
+	ptr->long_state = (uint8_t*)mmap(0, hashMemSize, PROT_READ | PROT_WRITE,
+		MAP_PRIVATE | MAP_ANON, -1, 0);
 #else
 	ptr->long_state = (uint8_t*)mmap(0, hashMemSize, PROT_READ | PROT_WRITE,
 		MAP_PRIVATE | MAP_ANONYMOUS | MAP_HUGETLB | MAP_POPULATE, 0, 0);

--- a/xmrstak/backend/cpu/minethd.cpp
+++ b/xmrstak/backend/cpu/minethd.cpp
@@ -84,7 +84,7 @@ bool minethd::thd_setaffinity(std::thread::native_handle_type h, uint64_t cpu_id
 	CPU_ZERO(&mn);
 	CPU_SET(cpu_id, &mn);
 	return pthread_setaffinity_np(h, sizeof(cpuset_t), &mn) == 0;
-#else
+#elif !defined(__OpenBSD__)
 	cpu_set_t mn;
 	CPU_ZERO(&mn);
 	CPU_SET(cpu_id, &mn);

--- a/xmrstak/backend/cpu/minethd.cpp
+++ b/xmrstak/backend/cpu/minethd.cpp
@@ -84,7 +84,9 @@ bool minethd::thd_setaffinity(std::thread::native_handle_type h, uint64_t cpu_id
 	CPU_ZERO(&mn);
 	CPU_SET(cpu_id, &mn);
 	return pthread_setaffinity_np(h, sizeof(cpuset_t), &mn) == 0;
-#elif !defined(__OpenBSD__)
+#elif defined(__OpenBSD__)
+        printer::inst()->print_msg(L0,"WARNING: thread pinning is not supported under OPENBSD.");
+#else
 	cpu_set_t mn;
 	CPU_ZERO(&mn);
 	CPU_SET(cpu_id, &mn);

